odata-csdl/1 Introduction.md

```diff
@@ -488,6 +488,306 @@ representation of primitive type values in URLs and
+[`Measures.DurationGranularity`](https://github.com/oasis-tcs/odata-vocabularies/blob/master/vocabularies/Org.OData.Measures.V1.md#DurationGranularity),
+see [OData-VocMeasures](#ODataVocMeasures).
+
+::: {.varjson .rep}
+### ##subisec `$Precision`
+
+The value of `$Precision` is a number.
+
+Absence of `$Precision` means arbitrary precision.
+:::
+
+::: {.varjson .example}
+Example ##ex: `Precision` facet applied to the `DateTimeOffset` type
+```json
+"SuggestedTimes": {
+  "$Type": "Edm.DateTimeOffset",
+  "$Collection": true,
+  "$Precision": 6
+}
+```
+:::
+
+::: {.varxml .rep}
+### ##subisec Attribute `Precision`
+
+The value of `Precision` is a number.
+
+If not specified for a decimal value, the decimal value has
+arbitrary precision.
+
+If not specified for a temporal value, the temporal value has a
+precision of zero.
+:::
+
+::: {.varxml .example}
+Example ##ex: [`Precision`](#Precision) facet applied to the
+`DateTimeOffset` type
+```xml
+<Property Name="SuggestedTimes" Type="Collection(Edm.DateTimeOffset)"
+          Precision="6" />
+```
+:::
+
+### ##subsubsec Scale
+
+A non-negative integer value specifying the maximum number of digits
+allowed to the right of the decimal point, or one of the symbolic values
+`floating` or `variable`.
+
+The value `floating` means that the decimal value represents a
+decimal floating-point number whose number of significant digits is the
+value of the [`Precision`](#Precision) facet. OData 4.0 responses MUST
+NOT specify the value `floating`.
+
+The value `variable` means that the number of digits to the right of the
+decimal point can vary from zero to the value of the
+[`Precision`](#Precision) facet.
+
+An integer value means that the number of digits to the right of the
+decimal point may vary from zero to the value of the `Scale` facet, and
+the number of digits to the left of the decimal point may vary from one
+to the value of the `Precision` facet minus the value of the `Scale`
+facet. If `Precision` is equal to `Scale`, a single zero MUST precede
+the decimal point.
+
+The value of `Scale` MUST be less than or equal to the value of
+[`Precision`](#Precision).
+
+Note: if the underlying data store allows negative scale, services may
+use a [`Precision`](#Precision) with the absolute value of the negative
+scale added to the actual number of significant decimal digits, and
+client-provided values may have to be rounded before being stored.
+
+::: {.varjson .rep}
+### ##subisec `$Scale`
+
+The value of `$Scale` is a number or a string with one of the symbolic
+values `floating` or `variable`.
+
+Services SHOULD use lower-case values; clients SHOULD accept values in a
+case-insensitive manner.
+
+Absence of `$Scale` means `variable`.
+:::
+
+::: {.varjson .example}
+Example ##ex: [`Precision`](#Precision)`=3` and `Scale=2`.  
+Allowed values: 1.23, 0.23, 3.14 and 0.7, not allowed values: 123, 12.3
+```json
+"Amount32": {
+  "$Nullable": true,
```

Comment:
Nullable is introduced only in section 7.2, can we avoid it in these early examples?

Reply:
Well, if we want to have isomorphic examples for JSON and XML, then one of them has to talk about Nullable because they have opposite defaults.

Where do we want to have Nullable: in JSON or in XML?

Reply:
Leave the examples as they are but add a comment: "The `$Nullable` member is explained in section ...".